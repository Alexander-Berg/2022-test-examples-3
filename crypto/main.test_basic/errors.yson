{
    "error" = "TNode type uint64_node expected, actual type null, during getting key=UserID";
};
