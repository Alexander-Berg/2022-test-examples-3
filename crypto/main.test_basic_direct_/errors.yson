{
    "error" = "[Lal id = 6] Invalid lal id";
};
{
    "error" = "[Lal id = 7] Invalid lal id";
};
{
    "error" = "[Lal id = 1500000008] Unsupported parent type: SpecialGoal";
};
{
    "error" = "[Lal id = 1500000015] Unknown parent type: unknown_parent_type";
};
