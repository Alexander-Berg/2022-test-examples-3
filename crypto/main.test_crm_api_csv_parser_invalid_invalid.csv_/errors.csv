{"object"={"create_date_time"="2020-01-01";"id"="11";"client_uniq_id"="111";"client_ids"="1,4,7";"emails"="xxx@local.local,x.x.x@gmail.com";"phones"="xxx,81234567890";"emails_md5"="3c56ff8fef0f6c65b36b2d25720fe276,8D2B135BA9BD88FBFBD8ECECB31196D1";"phones_md5"="C3281A42C6EF3CBAF3508F9738381816,5a893224859c93b3be4c1be4389dc58f";"order_status"="IN_PROGRESS";"revenue"="11.00";"cost"="44.44";};"what"="Invalid 'phone' identifier 'xxx'";"line"=2;};
{"object"={"create_date_time"="2020-02-02";"id"="22";"client_uniq_id"="222";"client_ids"="1,4,7";"emails"="i.i.i@gmail.com";"phones"="80987654321";"emails_md5"="3c56ff8fef0f6c65b36b2d25720fe276,8D2B135BA9BD88FBFBD8ECECB31196D1";"phones_md5"="C3281A42C6EF3CBAF3508F9738381816,5a893224859c93b3be4c1be4389dc58f";"order_status"="XXX";"revenue"="11.00";"cost"="44.44";};"what"="Invalid 'order_status' value 'XXX'";"line"=3;};
