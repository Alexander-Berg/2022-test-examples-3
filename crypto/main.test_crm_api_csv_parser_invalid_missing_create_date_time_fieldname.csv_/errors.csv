{"object"="missing_create_date_time_fieldname.csv";"what"="Missing 'create_date_time' field name";"line"=1;};
