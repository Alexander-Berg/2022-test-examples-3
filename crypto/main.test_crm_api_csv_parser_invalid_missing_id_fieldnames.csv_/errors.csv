{"object"="missing_id_fieldnames.csv";"what"="Missing field names for identifiers";"line"=1;};
