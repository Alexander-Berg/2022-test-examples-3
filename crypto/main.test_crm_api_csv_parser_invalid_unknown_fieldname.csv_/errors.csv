{"object"="unknown_fieldname.csv";"what"="Field names contain unknown field 'xxx'";"line"=1;};
