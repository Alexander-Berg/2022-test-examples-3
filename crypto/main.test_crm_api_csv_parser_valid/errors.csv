{"object"={"create_date_time"="2020-02-02";"id"="22";"client_uniq_id"="222";"client_ids"="1,4,7";"emails"="i.i.i@gmail.com";"phones"="80987654321";"emails_md5"="3c56ff8fef0f6c65b36b2d25720fe276,8D2B135BA9BD88FBFBD8ECECB31196D1";"phones_md5"="C3281A42C6EF3CBAF3508F9738381816,5a893224859c93b3be4c1be4389dc58f";"order_status"="XXX";"revenue"="11.00";"cost"="44.44";};"what"="Invalid 'order_status' value 'XXX'";"line"=4;};
{"object"={"create_date_time"="2020-01-01 23:59:58";"id"="33";"client_uniq_id"="333";"client_ids"="1,4,7";"emails"="xxx@local.local";"phones"="";"emails_md5"="";"phones_md5"="";"order_status"="IN_PROGRESS";"revenue"="33.00";"cost"="44.44";};"what"="Row is too old";"line"=5;};
