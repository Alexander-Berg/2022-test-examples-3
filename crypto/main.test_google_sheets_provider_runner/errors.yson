{
    "Value" = {
        "Yclid" = 333;
    };
    "ErrorMsg" = "Missing column 'Conversion Time'";
};
