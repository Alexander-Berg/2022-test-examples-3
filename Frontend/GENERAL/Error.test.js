import React from 'react';
import { mount } from 'enzyme';

import { Error } from './Error';

describe('Should render', () => {
    it('error 1', () => {
        const wrapper = mount(
            <Error
                data={{
                    message: {
                        ru: 'Текст ru message',
                        en: 'Текст en message',
                    },
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 2', () => {
        const wrapper = mount(
            <Error
                data={{ detail: 'Текст detail' }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 3', () => {
        const wrapper = mount(
            <Error
                data={{
                    message: {
                        ru: 'Текст ru message',
                        en: 'Текст en message',
                    },
                    title: {
                        ru: 'Текст ru title',
                        en: 'Текст en title',
                    },
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 4', () => {
        const wrapper = mount(
            <Error
                data={{
                    detail: 'Текст detail',
                    title: {
                        ru: 'Текст ru title',
                        en: 'Текст en title',
                    },
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 5', () => {
        const wrapper = mount(
            <Error
                data={{
                    code: 'validation_error',
                    detail: 'Текст detail',
                    extra: {
                        __all__: ['Такое себе'],
                        foo: ['Раз', 'Два'],
                    },
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 6', () => {
        const wrapper = mount(
            <Error
                data={{
                    code: 'validation_error',
                    detail: 'Текст detail',
                    extra: {
                        foo: ['Раз', 'Два'],
                    },
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 7', () => {
        const wrapper = mount(
            <Error
                data={{
                    code: 'validation_error',
                    detail: 'Текст detail',
                    extra: {
                        foo: {
                            ru: ['Раз', 'Два'],
                        },
                    },
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 8', () => {
        const wrapper = mount(
            <Error
                data={{
                    message: 'None',
                    code: 'YAWF_VALIDATION_ERROR',
                    params: {
                        contacts: [
                            {
                                non_field_errors: ['Неверное значение контакта 0'],
                            },
                            {},
                            {
                                non_field_errors: ['Неверное значение контакта 2'],
                            },
                        ],
                    },
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 9', () => {
        const wrapper = mount(
            <Error
                data={{
                    message: 'String message',
                    title: 'String title',
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 10', () => {
        const wrapper = mount(
            <Error
                data={{
                    extra: {
                        path: [
                            'Узел в дереве ролей не найден',
                        ],
                    },
                    message: 'Invalid data sent',
                    code: 'error',
                    detail: 'Произошла ошибка сервера.',
                    title: 'Ошибка IDM',
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 11', () => {
        const wrapper = mount(
            <Error
                data={{
                    message: '',
                    code: 'integration-error',
                    detail: 'IDM did not fulfil the request.',
                    extra: {
                        raw: '{"error_code": "READONLY_STATE", "message": "IDM is in a read-only state, cannot handle write intending request"}',
                        message: 'IDM is in a read-only state, cannot handle write intending request',
                        errors: null,
                    },
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 12: dispenser error with data object', () => {
        const wrapper = mount(
            <Error
                data={{
                    message: '',
                    code: 'integration-error',
                    detail: 'Dispenser did not fulfil the request.',
                    description: 'Dispenser is in a read-only state, cannot handle write intending request',
                }}
                type="dispenser"
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 13: dispenser error with string data', () => {
        const wrapper = mount(
            <Error
                data="Dispenser is in a read-only state, cannot handle write intending request"
                type="dispenser"
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('error 14', () => {
        const wrapper = mount(
            <Error
                data={{
                    message: '',
                    code: 'integration-error',
                    detail: 'IDM did not fulfil the request.',
                    description: 'IDM is in a read-only state, cannot handle write intending request',
                }}
                type="dispenser"
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });
});
