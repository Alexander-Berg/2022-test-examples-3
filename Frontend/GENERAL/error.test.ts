import debugFactory, { IDebugger } from 'debug';
import { Request, Response } from 'express';

import ErrorController from './error';

import * as errors from './errors';

function getMockedResponse() {
    return {
        status: jest.fn().mockReturnThis(),
        json: jest.fn().mockReturnThis(),
        end: jest.fn(),
    } as unknown as Response;
}

describe('ErrorController', () => {
    let debug: IDebugger;

    beforeEach(() => {
        debug = debugFactory('test');
    });

    describe('.getHandler', () => {
        it('should return function', () => {
            const controller = new ErrorController({ debug });

            const expected = 'function';

            const actual = controller.getHandler();

            expect(typeof actual).toEqual(expected);
        });

        it('should response with status 500 when handled unexpected error', () => {
            const controller = new ErrorController({ debug });

            const fakeError = new Error('unknown error');
            const fakeRequest = { id: '123' } as unknown as Request;
            const fakeResponse = getMockedResponse();
            const fakeNext = jest.fn();

            const handler = controller.getHandler();

            handler(fakeError, fakeRequest, fakeResponse, fakeNext);

            expect(fakeResponse.status).toHaveBeenCalledWith(500);
            expect(fakeResponse.json).not.toHaveBeenCalled();
            expect(fakeResponse.end).toHaveBeenCalledWith();

            expect(fakeNext).not.toHaveBeenCalled();
        });

        it('should response with status 500 when handled error with typeof HttpResponseError', () => {
            const controller = new ErrorController({ debug });

            const fakeError = new errors.HttpResponseError();
            const fakeRequest = { id: '123' } as unknown as Request;
            const fakeResponse = getMockedResponse();
            const fakeNext = jest.fn();

            const handler = controller.getHandler();

            const expectedPayload = {
                comment: 'Internal Server Error',
                message: '',
            };

            handler(fakeError, fakeRequest, fakeResponse, fakeNext);

            expect(fakeResponse.status).toHaveBeenCalledWith(500);
            expect(fakeResponse.json).toHaveBeenCalledWith(expectedPayload);
            expect(fakeResponse.end).toHaveBeenCalledWith();

            expect(fakeNext).not.toHaveBeenCalled();
        });

        it('should response with status from HttpResponseError-based error', () => {
            const controller = new ErrorController({ debug });

            const fakeError = new errors.InvalidPayloadError('passed payload is invalid for some reason');
            const fakeRequest = { id: '123' } as unknown as Request;
            const fakeResponse = getMockedResponse();
            const fakeNext = jest.fn();

            const handler = controller.getHandler();

            const expectedPayload = {
                comment: 'Invalid Payload',
                message: 'passed payload is invalid for some reason',
            };

            handler(fakeError, fakeRequest, fakeResponse, fakeNext);

            expect(fakeResponse.status).toHaveBeenCalledWith(406);
            expect(fakeResponse.json).toHaveBeenCalledWith(expectedPayload);
            expect(fakeResponse.end).toHaveBeenCalledWith();

            expect(fakeNext).not.toHaveBeenCalled();
        });
    });
});
