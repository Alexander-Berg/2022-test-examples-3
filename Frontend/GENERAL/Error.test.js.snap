// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render error 1 1`] = `
<Error
  data={
    Object {
      "message": Object {
        "en": "Текст en message",
        "ru": "Текст ru message",
      },
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "message": Object {
            "en": "Текст en message",
            "ru": "Текст ru message",
          },
        }
      }
      key="title-place"
    />
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "message": Object {
            "en": "Текст en message",
            "ru": "Текст ru message",
          },
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        Текст ru message
      </div>
    </MessagePlace>
  </div>
</Error>
`;

exports[`Should render error 2 1`] = `
<Error
  data={
    Object {
      "detail": "Текст detail",
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "detail": "Текст detail",
        }
      }
      key="title-place"
    />
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "detail": "Текст detail",
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        Текст detail
      </div>
    </MessagePlace>
  </div>
</Error>
`;

exports[`Should render error 3 1`] = `
<Error
  data={
    Object {
      "message": Object {
        "en": "Текст en message",
        "ru": "Текст ru message",
      },
      "title": Object {
        "en": "Текст en title",
        "ru": "Текст ru title",
      },
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "message": Object {
            "en": "Текст en message",
            "ru": "Текст ru message",
          },
          "title": Object {
            "en": "Текст en title",
            "ru": "Текст ru title",
          },
        }
      }
      key="title-place"
    >
      <div
        className="Error-Title Error-Title_0_E Error-Title_1_r Error-Title_2_r Error-Title_3_o Error-Title_4_r Error-Title_5_- Error-Title_6_T Error-Title_7_i Error-Title_8_t Error-Title_9_l Error-Title_10_e Error-Title_11_P Error-Title_12_l Error-Title_13_a Error-Title_14_c Error-Title_15_e"
      >
        Текст ru title
      </div>
    </TitlePlace>
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "message": Object {
            "en": "Текст en message",
            "ru": "Текст ru message",
          },
          "title": Object {
            "en": "Текст en title",
            "ru": "Текст ru title",
          },
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        Текст ru message
      </div>
    </MessagePlace>
  </div>
</Error>
`;

exports[`Should render error 4 1`] = `
<Error
  data={
    Object {
      "detail": "Текст detail",
      "title": Object {
        "en": "Текст en title",
        "ru": "Текст ru title",
      },
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "detail": "Текст detail",
          "title": Object {
            "en": "Текст en title",
            "ru": "Текст ru title",
          },
        }
      }
      key="title-place"
    >
      <div
        className="Error-Title Error-Title_0_E Error-Title_1_r Error-Title_2_r Error-Title_3_o Error-Title_4_r Error-Title_5_- Error-Title_6_T Error-Title_7_i Error-Title_8_t Error-Title_9_l Error-Title_10_e Error-Title_11_P Error-Title_12_l Error-Title_13_a Error-Title_14_c Error-Title_15_e"
      >
        Текст ru title
      </div>
    </TitlePlace>
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "detail": "Текст detail",
          "title": Object {
            "en": "Текст en title",
            "ru": "Текст ru title",
          },
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        Текст detail
      </div>
    </MessagePlace>
  </div>
</Error>
`;

exports[`Should render error 5 1`] = `
<Error
  data={
    Object {
      "code": "validation_error",
      "detail": "Текст detail",
      "extra": Object {
        "__all__": Array [
          "Такое себе",
        ],
        "foo": Array [
          "Раз",
          "Два",
        ],
      },
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "code": "validation_error",
          "detail": "Текст detail",
          "extra": Object {
            "__all__": Array [
              "Такое себе",
            ],
            "foo": Array [
              "Раз",
              "Два",
            ],
          },
        }
      }
      key="title-place"
    />
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "code": "validation_error",
          "detail": "Текст detail",
          "extra": Object {
            "__all__": Array [
              "Такое себе",
            ],
            "foo": Array [
              "Раз",
              "Два",
            ],
          },
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        Текст detail
      </div>
    </MessagePlace>
    <Field
      className="Error-Field"
      key="extra-__all__"
      name="__all__"
      values={
        Array [
          "Такое себе",
        ]
      }
    >
      <div
        className="Error-Field"
      >
        <div
          className="Error-Value"
          key="value-0"
        >
          Такое себе
        </div>
      </div>
    </Field>
    <Field
      className="Error-Field"
      key="extra-foo"
      name="foo"
      values={
        Array [
          "Раз",
          "Два",
        ]
      }
    >
      <div
        className="Error-Field"
      >
        <div
          className="Error-Title"
          key="title"
        >
          foo
        </div>
        <div
          className="Error-Value"
          key="value-0"
        >
          Раз
        </div>
        <div
          className="Error-Value"
          key="value-1"
        >
          Два
        </div>
      </div>
    </Field>
  </div>
</Error>
`;

exports[`Should render error 6 1`] = `
<Error
  data={
    Object {
      "code": "validation_error",
      "detail": "Текст detail",
      "extra": Object {
        "foo": Array [
          "Раз",
          "Два",
        ],
      },
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "code": "validation_error",
          "detail": "Текст detail",
          "extra": Object {
            "foo": Array [
              "Раз",
              "Два",
            ],
          },
        }
      }
      key="title-place"
    />
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "code": "validation_error",
          "detail": "Текст detail",
          "extra": Object {
            "foo": Array [
              "Раз",
              "Два",
            ],
          },
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        Текст detail
      </div>
    </MessagePlace>
    <Field
      className="Error-Field"
      key="extra-foo"
      name="foo"
      values={
        Array [
          "Раз",
          "Два",
        ]
      }
    >
      <div
        className="Error-Field"
      >
        <div
          className="Error-Title"
          key="title"
        >
          foo
        </div>
        <div
          className="Error-Value"
          key="value-0"
        >
          Раз
        </div>
        <div
          className="Error-Value"
          key="value-1"
        >
          Два
        </div>
      </div>
    </Field>
  </div>
</Error>
`;

exports[`Should render error 7 1`] = `
<Error
  data={
    Object {
      "code": "validation_error",
      "detail": "Текст detail",
      "extra": Object {
        "foo": Object {
          "ru": Array [
            "Раз",
            "Два",
          ],
        },
      },
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "code": "validation_error",
          "detail": "Текст detail",
          "extra": Object {
            "foo": Object {
              "ru": Array [
                "Раз",
                "Два",
              ],
            },
          },
        }
      }
      key="title-place"
    />
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "code": "validation_error",
          "detail": "Текст detail",
          "extra": Object {
            "foo": Object {
              "ru": Array [
                "Раз",
                "Два",
              ],
            },
          },
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        Текст detail
      </div>
    </MessagePlace>
    <Field
      className="Error-Field"
      key="extra-foo.ru"
      name="foo.ru"
      values={
        Array [
          "Раз",
          "Два",
        ]
      }
    >
      <div
        className="Error-Field"
      >
        <div
          className="Error-Title"
          key="title"
        >
          foo.ru
        </div>
        <div
          className="Error-Value"
          key="value-0"
        >
          Раз
        </div>
        <div
          className="Error-Value"
          key="value-1"
        >
          Два
        </div>
      </div>
    </Field>
  </div>
</Error>
`;

exports[`Should render error 8 1`] = `
<Error
  data={
    Object {
      "code": "YAWF_VALIDATION_ERROR",
      "message": "None",
      "params": Object {
        "contacts": Array [
          Object {
            "non_field_errors": Array [
              "Неверное значение контакта 0",
            ],
          },
          Object {},
          Object {
            "non_field_errors": Array [
              "Неверное значение контакта 2",
            ],
          },
        ],
      },
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "code": "YAWF_VALIDATION_ERROR",
          "message": "None",
          "params": Object {
            "contacts": Array [
              Object {
                "non_field_errors": Array [
                  "Неверное значение контакта 0",
                ],
              },
              Object {},
              Object {
                "non_field_errors": Array [
                  "Неверное значение контакта 2",
                ],
              },
            ],
          },
        }
      }
      key="title-place"
    >
      <div
        className="Error-Title Error-Title_0_E Error-Title_1_r Error-Title_2_r Error-Title_3_o Error-Title_4_r Error-Title_5_- Error-Title_6_T Error-Title_7_i Error-Title_8_t Error-Title_9_l Error-Title_10_e Error-Title_11_P Error-Title_12_l Error-Title_13_a Error-Title_14_c Error-Title_15_e"
      >
        YAWF_VALIDATION_ERROR
      </div>
    </TitlePlace>
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "code": "YAWF_VALIDATION_ERROR",
          "message": "None",
          "params": Object {
            "contacts": Array [
              Object {
                "non_field_errors": Array [
                  "Неверное значение контакта 0",
                ],
              },
              Object {},
              Object {
                "non_field_errors": Array [
                  "Неверное значение контакта 2",
                ],
              },
            ],
          },
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        None
      </div>
    </MessagePlace>
    <Field
      className="Error-Field"
      key="params-contacts.0.non_field_errors"
      name="contacts.0.non_field_errors"
      values={
        Array [
          "Неверное значение контакта 0",
        ]
      }
    >
      <div
        className="Error-Field"
      >
        <div
          className="Error-Title"
          key="title"
        >
          contacts.0.non_field_errors
        </div>
        <div
          className="Error-Value"
          key="value-0"
        >
          Неверное значение контакта 0
        </div>
      </div>
    </Field>
    <Field
      className="Error-Field"
      key="params-contacts.2.non_field_errors"
      name="contacts.2.non_field_errors"
      values={
        Array [
          "Неверное значение контакта 2",
        ]
      }
    >
      <div
        className="Error-Field"
      >
        <div
          className="Error-Title"
          key="title"
        >
          contacts.2.non_field_errors
        </div>
        <div
          className="Error-Value"
          key="value-0"
        >
          Неверное значение контакта 2
        </div>
      </div>
    </Field>
  </div>
</Error>
`;

exports[`Should render error 9 1`] = `
<Error
  data={
    Object {
      "message": "String message",
      "title": "String title",
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "message": "String message",
          "title": "String title",
        }
      }
      key="title-place"
    >
      <div
        className="Error-Title Error-Title_0_E Error-Title_1_r Error-Title_2_r Error-Title_3_o Error-Title_4_r Error-Title_5_- Error-Title_6_T Error-Title_7_i Error-Title_8_t Error-Title_9_l Error-Title_10_e Error-Title_11_P Error-Title_12_l Error-Title_13_a Error-Title_14_c Error-Title_15_e"
      >
        String title
      </div>
    </TitlePlace>
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "message": "String message",
          "title": "String title",
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        String message
      </div>
    </MessagePlace>
  </div>
</Error>
`;

exports[`Should render error 10 1`] = `
<Error
  data={
    Object {
      "code": "error",
      "detail": "Произошла ошибка сервера.",
      "extra": Object {
        "path": Array [
          "Узел в дереве ролей не найден",
        ],
      },
      "message": "Invalid data sent",
      "title": "Ошибка IDM",
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "code": "error",
          "detail": "Произошла ошибка сервера.",
          "extra": Object {
            "path": Array [
              "Узел в дереве ролей не найден",
            ],
          },
          "message": "Invalid data sent",
          "title": "Ошибка IDM",
        }
      }
      key="title-place"
    >
      <div
        className="Error-Title Error-Title_0_E Error-Title_1_r Error-Title_2_r Error-Title_3_o Error-Title_4_r Error-Title_5_- Error-Title_6_T Error-Title_7_i Error-Title_8_t Error-Title_9_l Error-Title_10_e Error-Title_11_P Error-Title_12_l Error-Title_13_a Error-Title_14_c Error-Title_15_e"
      >
        Ошибка IDM
      </div>
    </TitlePlace>
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "code": "error",
          "detail": "Произошла ошибка сервера.",
          "extra": Object {
            "path": Array [
              "Узел в дереве ролей не найден",
            ],
          },
          "message": "Invalid data sent",
          "title": "Ошибка IDM",
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        Invalid data sent
      </div>
    </MessagePlace>
    <Field
      className="Error-Field"
      key="extra-path"
      name="path"
      values={
        Array [
          "Узел в дереве ролей не найден",
        ]
      }
    >
      <div
        className="Error-Field"
      >
        <div
          className="Error-Title"
          key="title"
        >
          path
        </div>
        <div
          className="Error-Value"
          key="value-0"
        >
          Узел в дереве ролей не найден
        </div>
      </div>
    </Field>
  </div>
</Error>
`;

exports[`Should render error 11 1`] = `
<Error
  data={
    Object {
      "code": "integration-error",
      "detail": "IDM did not fulfil the request.",
      "extra": Object {
        "errors": null,
        "message": "IDM is in a read-only state, cannot handle write intending request",
        "raw": "{\\"error_code\\": \\"READONLY_STATE\\", \\"message\\": \\"IDM is in a read-only state, cannot handle write intending request\\"}",
      },
      "message": "",
    }
  }
>
  <div
    className="Error"
  >
    <TitlePlace
      className="Error-TitlePlace"
      data={
        Object {
          "code": "integration-error",
          "detail": "IDM did not fulfil the request.",
          "extra": Object {
            "errors": null,
            "message": "IDM is in a read-only state, cannot handle write intending request",
            "raw": "{\\"error_code\\": \\"READONLY_STATE\\", \\"message\\": \\"IDM is in a read-only state, cannot handle write intending request\\"}",
          },
          "message": "",
        }
      }
      key="title-place"
    />
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "code": "integration-error",
          "detail": "IDM did not fulfil the request.",
          "extra": Object {
            "errors": null,
            "message": "IDM is in a read-only state, cannot handle write intending request",
            "raw": "{\\"error_code\\": \\"READONLY_STATE\\", \\"message\\": \\"IDM is in a read-only state, cannot handle write intending request\\"}",
          },
          "message": "",
        }
      }
      key="message-place"
      type="abc"
    >
      <div
        className="Error-MessagePlace"
      >
        i18n:integration-error
      </div>
    </MessagePlace>
    <Field
      className="Error-Field"
      key="extra-errors"
      name="errors"
      values={
        Array [
          "null",
        ]
      }
    >
      <div
        className="Error-Field"
      >
        <div
          className="Error-Title"
          key="title"
        >
          errors
        </div>
        <div
          className="Error-Value"
          key="value-0"
        >
          null
        </div>
      </div>
    </Field>
    <Field
      className="Error-Field"
      key="extra-message"
      name="message"
      values={
        Array [
          "IDM is in a read-only state, cannot handle write intending request",
        ]
      }
    >
      <div
        className="Error-Field"
      >
        <div
          className="Error-Title"
          key="title"
        >
          message
        </div>
        <div
          className="Error-Value"
          key="value-0"
        >
          IDM is in a read-only state, cannot handle write intending request
        </div>
      </div>
    </Field>
  </div>
</Error>
`;

exports[`Should render error 12: dispenser error with data object 1`] = `
<Error
  data={
    Object {
      "code": "integration-error",
      "description": "Dispenser is in a read-only state, cannot handle write intending request",
      "detail": "Dispenser did not fulfil the request.",
      "message": "",
    }
  }
  type="dispenser"
>
  <div
    className="Error"
  >
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "code": "integration-error",
          "description": "Dispenser is in a read-only state, cannot handle write intending request",
          "detail": "Dispenser did not fulfil the request.",
          "message": "",
        }
      }
      key="message-place"
      type="dispenser"
    >
      <div
        className="Error-MessagePlace"
      >
        Dispenser is in a read-only state, cannot handle write intending request
      </div>
    </MessagePlace>
  </div>
</Error>
`;

exports[`Should render error 13: dispenser error with string data 1`] = `
<Error
  data="Dispenser is in a read-only state, cannot handle write intending request"
  type="dispenser"
>
  <div
    className="Error"
  >
    <MessagePlace
      className="Error-MessagePlace"
      data="Dispenser is in a read-only state, cannot handle write intending request"
      key="message-place"
      type="dispenser"
    >
      <div
        className="Error-MessagePlace"
      >
        Dispenser is in a read-only state, cannot handle write intending request
      </div>
    </MessagePlace>
  </div>
</Error>
`;

exports[`Should render error 14 1`] = `
<Error
  data={
    Object {
      "code": "integration-error",
      "description": "IDM is in a read-only state, cannot handle write intending request",
      "detail": "IDM did not fulfil the request.",
      "message": "",
    }
  }
  type="dispenser"
>
  <div
    className="Error"
  >
    <MessagePlace
      className="Error-MessagePlace"
      data={
        Object {
          "code": "integration-error",
          "description": "IDM is in a read-only state, cannot handle write intending request",
          "detail": "IDM did not fulfil the request.",
          "message": "",
        }
      }
      key="message-place"
      type="dispenser"
    >
      <div
        className="Error-MessagePlace"
      >
        IDM is in a read-only state, cannot handle write intending request
      </div>
    </MessagePlace>
  </div>
</Error>
`;
